R/Utils.R

```diff
@@ -196,4 +198,74 @@ SeuratToAnnData <- function(seuratObj, outFileBaseName, assayName = NULL, doDiet
   }
 
   return(ret)
+}
+
+#' @title Remove UCell Doublets
+#'
+#' @description Attempts to remove dump-gate "doublets" from the main three cell type subset objects (Tcell, Bcell, MoMacDC)
+#' @param parentUCell The subset's target cell type's UCell score. (Computed by scGate early in processing).
+#' @param otherUCellScores A vector of cell type UCell scores to attempt to discard from downstream processing. By default, these are cell types that comprise the dump gate in RIRA's scGate model. 
+#' @param clusterResolution The metadata column name that defines cluster membership (e.g. ClusterNames_0.2)
+#' @param cosineSimilarityCutoff A number between -1 and 1 that is used to threshold/gate the prospective doublets as defined by the computed score.
+#' @return Returns a seurat object with the UCell doublets removed. 
+#' @export
+#' 
+.RemoveUCellDoublets <- function(seuratObj, parentUCell = NULL, otherUCellScores = c("Platelet_MK_UCell", "Erythrocyte_UCell", "Neutrophil_UCell", "Epithelial_UCell", "Stromal_UCell"), clusterResolution = "ClusterNames_1.2", cosineSimilarityCutoff = 0.5){
+  #test for valid parentUCell
+  if(is.null(parentUCell)){
```

Comment:
I'm all for having some method where we need to specify the parent. However, should we have some RIRA method that perhaps attempts to look at the RIRA/typist call, and maps this to the right parent?

Reply:
yeah, I think stitching this function to the tail end of Classify_ImmuneCells() seems natural, so maybe we just read in the majority_voting/predicted_labels column from there?